theano/gof/opt.py

```diff
@@ -1261,10 +1289,72 @@ def tracks(self):
         return t
 
     def transform(self, node):
-        for opt in self.opts:
-            repl = opt.transform(node)
-            if repl:
-                return repl
+        if len(self.opts) == 0:
+            return
+
+        def apply_mult_opts(node, fgraph, multiple_opts=False, prev_repl=None):
+            opts = self.track_map[type(node.op)] + self.track_map[node.op] + self.track_map[None]
+            repl = prev_repl
+            for opt in opts:
+                opt_start = time.time()
+                repl = opt.transform(node)
+                opt_finish = time.time()
+                self.time_opts[opt] += opt_start - opt_finish
+                self.process_count[opt] += 1
+                if not repl:
+                    continue
+                else:
+                    self.node_created[opt] += len(graph.ops(fgraph.variables, repl))
+                    self.applied_true[opt] += 1
+                    if not multiple_opts or not repl[0].owner:
+                        return repl
+                    assert len(repl) == 1
+                    # Ensuring not the input of graph
+                    assert repl[0].owner
+                    new_node = repl[0].owner
+                    apply_mult_opts(new_node, fgraph, True, repl)
```

Comment:
This won't change repl. So the second optimization if it apply won't be applied. Can you fix this problem in the next few hours? I would like to merge this PR soon to go to the next phase.